### PR TITLE
Theme fixes/updates, deprecate Drawer

### PR DESCRIPTION
**Pull request by GroovinChip**
This pull request contains:

- Working dark theme: The problem with the dark theme in the source is `scaffoldBackgroundColor: primaryColor`, since the `primaryColor` was defined as `Colors.white`. This has been commented out.
- Update Dark theme colors to not be true black: true black in this app makes content hard to differentiate.
- Deprecate the `Drawer`: It is my opinion that the Navigation Drawer is visually and (mostly) functionally outdated. I believe the `modalBottomSheet` provides a more modern and better user experience.

### CODE REVIEW

File: .flutter-plugins-dependencies [hunk 1]
```diff
@@ -0,0 +1 @@
...lang.org\\\\sqflite-1.1.5\\\\","dependencies":[]},{"name":"url_launcher","path":"C:\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\url_launcher-5.0.2\\\\","dependencies":[]},{"name":"webview_flutter","path":"C:\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\webview_flutter-0.3.7+1\\\\","dependencies":[]}],"android":[{"name":"shared_preferences","path":"C:\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\shared_preferences-0.5.2+1\\\\","dependencies":[]},{"name":"sqflite","path":"C:\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\sqflite-1.1.5\\\\","dependencies":[]},{"name":"url_launcher","path":"C:\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\url_launcher-5.0.2\\\\","dependencies":[]},{"name":"webview_flutter","path":"C:\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\webview_flutter-0.3.7+1\\\\","dependencies":[]}],"macos":[],"linux":[],"windows":[],"web":[]},"dependencyGraph":[{"name":"shared_preferences","dependencies":[]},{"name":"sqflite","dependencies":[]},{"name":"url_launcher","dependencies":[]},{"name":"webview_flutter","dependencies":[]}],"date_created":"2020-01-31 23:28:13.704632","version":"1.14.4-pre.27"}
```
Comment (filiph) [inline .flutter-plugins-dependencies]: Oops, looks like we have an out of date .gitignore. This should be in version control.

File: lib/main.dart [hunk 2]
```diff
@@ -45,15 +46,14 @@ class MyApp extends StatelessWidget {
       title: 'Flutter Demo',
       darkTheme: ThemeData.dark(),
       theme: ThemeData(
-          brightness: Provider.of<PrefsNotifier>(context).userDarkMode
-              ? Brightness.dark
-              : Brightness.light,
-          canvasColor: Theme.of(context).brightness == Brightness.dark ||
-                  Provider.of<PrefsNotifier>(context).userDarkMode
+          brightness:
+              Provider.of<PrefsNotifier>(context).userDarkMode ? Brightness.dark : Brightness.light,
```
Comment (filiph) [inline lib/main.dart:50]: Is this `dartfmt` for 80 columns? For consistency, it should.

File: lib/main.dart [hunk 3]
```diff
@@ -45,15 +46,14 @@ class MyApp extends StatelessWidget {
...
+          /*canvasColor: Theme.of(context).brightness == Brightness.dark ||
```
Comment (filiph) [inline lib/main.dart]: Prefer deleting instead of commenting out.

File: lib/main.dart [hunk 4]
```diff
@@ -143,7 +147,54 @@ class _MyHomePageState extends State<MyHomePage> {
                 ? LoadingInfo(loading)
                 : IconButton(
                     icon: Icon(Icons.menu),
-                    onPressed: () => Scaffold.of(context).openDrawer(),
+                    //onPressed: () => Scaffold.of(context).openDrawer(),
+                    onPressed: () => showModalBottomSheet(
```
Comment (filiph) [inline lib/main.dart:149]: [nit] Ideally, I'd like to see this as two separate PRs (the two changes seem orthogonal). But let's not complicate things if it's too much work weeding the two out.

File: lib/main.dart [hunk 5]
```diff
@@ -143,7 +147,54 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                      context: context,
+                      shape: RoundedRectangleBorder(
+                        borderRadius: BorderRadius.only(
+                          topLeft: Radius.circular(12),
+                          topRight: Radius.circular(12),
+                        ),
+                      ),
+                      builder: (context) => Container(
+                        padding: EdgeInsets.only(top: 8),
```
Comment (filiph) [inline lib/main.dart]: [nit] This and some other things here could be `const`. I don't think we are being consistent with this in `hn_app`, so no big deal.

File: lib/main.dart [hunk 6]
```diff
@@ -143,7 +147,54 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                            ListTile(
+                              leading: Icon(Icons.favorite),
+                              title: Text(
+                                'Favorites',
+                                style: TextStyle(
+                                  fontSize: 20,
+                                ),
+                              ),
+                              onTap: () {
+                                _pageNavigatorKey.currentState.pushReplacementNamed('/favorites');
```
Comment (filiph) [inline lib/main.dart]: Here we're getting a bit too much to the right. Might make sense to split into separate widgets, or even just methods.

File: lib/main.dart [hunk 7]
```diff
@@ -185,14 +238,13 @@ class _MyHomePageState extends State<MyHomePage> {
         ],
         onTap: (index) {
           _pageController.animateToPage(index,
-              duration: const Duration(milliseconds: 300),
-              curve: Curves.easeOutCubic);
+              duration: const Duration(milliseconds: 300), curve: Curves.easeOutCubic);
```
Comment (filiph) [inline lib/main.dart]: `dartfmt` here would make this whitespace-only change disappear.

File: lib/main.dart [hunk 8]
```diff
@@ -185,14 +238,13 @@ class _MyHomePageState extends State<MyHomePage> {
...
           setState(() {
             _currentIndex = index;
           });
         },
       ),
-      drawer: Drawer(
+      /*drawer: Drawer(
```
Comment (filiph) [inline lib/main.dart]: Remove.

File: lib/main.dart [hunk 9]
```diff
@@ -1,6 +1,7 @@
 import 'package:flutter/foundation.dart';
 import 'package:flutter/gestures.dart';
 import 'package:flutter/material.dart';
+import 'package:groovin_widgets/groovin_widgets.dart';
```
Comment (filiph) [inline lib/main.dart]: Question: what is this used for?

File: lib/src/widgets/search.dart [hunk 10]
```diff
@@ -32,17 +39,13 @@ class ArticleSearch extends SearchDelegate<Article> {
 
   @override
   Widget buildResults(BuildContext context) {
-    var results = articles
-        .where((a) => a.title.toLowerCase().contains(query.toLowerCase()));
+    var results = articles.where((a) => a.title.toLowerCase().contains(query.toLowerCase()));
```
Comment (filiph) [inline lib/src/widgets/search.dart]: Another file to be `dartfmt`'d (with 80 columns). You can run `dartfmt -w .` from the root of the project to format every Dart file at once.

**Review by filiph — CHANGES_REQUESTED** [review]
This is awesome, thank you!

Some minor changes needed, and a question or two. I would also love @efortuna to have a quick look (maybe after you address my feedback) — she's more familiar with at least some of the code you changed.

In any case, cheers!

File: .flutter-plugins-dependencies (hunk 1 above)
Comment (GroovinChip) [reply .flutter-plugins-dependencies]: I added `.flutter-plugins-dependencies` to the gitignore but `git status` still picks it up 🤔

File: lib/main.dart (hunk 2 above)
Comment (GroovinChip) [reply lib/main.dart:50]: I don't know, I just right-clicked in Android Studio and selected `dartfmt`

File: lib/main.dart (hunk 9 above)
Comment (GroovinChip) [reply lib/main.dart]: This is for the `ModalDrawerHandle()` widget used in the bottom sheet. It adds the little handle looking thing :)

File: lib/main.dart (hunk 4 above)
Comment (GroovinChip) [reply lib/main.dart:149]: Do you mean separating the theme changes and the widget changes? (like Drawer => BottomSheet?)

File: lib/main.dart (hunk 6 above)
Comment (GroovinChip) [reply lib/main.dart]: Sheet content extracted to it's own widget in `lib/widgets`

File: lib/main.dart (hunk 7 above)
Comment (GroovinChip) [reply lib/main.dart]: Yeah I don't know why it didn't do that 🤔

File: lib/main.dart (hunk 8 above)
Comment (GroovinChip) [reply lib/main.dart]: Removed.

File: lib/src/widgets/search.dart (hunk 10 above)
Comment (GroovinChip) [reply lib/src/widgets/search.dart]: `dartfmt` did nothing here again. Addressed manually.

File: lib/main.dart (hunk 7 above)
Comment (GroovinChip) [reply lib/main.dart]: addressed manually

File: lib/main.dart (hunk 2 above)
Comment (GroovinChip) [reply lib/main.dart:50]: addressed manually

File: lib/main.dart (hunk 5 above)
Comment (GroovinChip) [reply lib/main.dart]: adding it anyway :)
